presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/RecordingHiveMetastoreModule.java

```diff
@@ -28,24 +32,52 @@
 import static org.weakref.jmx.guice.ExportBinder.newExporter;
 
 public class RecordingHiveMetastoreModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     @Override
-    public void configure(Binder binder)
+    protected void setup(Binder binder)
     {
-        binder.bind(HiveMetastore.class)
-                .annotatedWith(ForCachingHiveMetastore.class)
-                .to(RecordingHiveMetastore.class)
-                .in(Scopes.SINGLETON);
-        binder.bind(RecordingHiveMetastore.class).in(Scopes.SINGLETON);
-        binder.bind(HiveBlockEncodingSerde.class).in(Scopes.SINGLETON);
+        if (buildConfigObject(HiveConfig.class).getRecordingPath() != null) {
+            install(new RecordingModule());
+        }
+        else {
+            install(new NoRecordingModule());
+        }
+    }
+
+    public static class RecordingModule
+            implements Module
+    {
+        @Override
+        public void configure(Binder binder)
+        {
+            binder.bind(HiveMetastore.class)
+                    .annotatedWith(ForCachingHiveMetastore.class)
+                    .to(RecordingHiveMetastore.class)
+                    .in(Scopes.SINGLETON);
+            binder.bind(RecordingHiveMetastore.class).in(Scopes.SINGLETON);
```

Comment:
Why exactly this one is needed?

Reply:
it is injected into `WriteHiveMetastoreRecordingProcedure`

Reply:
will add a test